jme3-android/src/main/java/com/jme3/audio/plugins/NativeVorbisFile.java

```diff
@@ -16,22 +57,69 @@ public class NativeVorbisFile {
     
     static {
         System.loadLibrary("decodejme");
-        nativeInit();
+        preInit();
     }
     
-    public NativeVorbisFile(int fd, long off, long len) throws IOException {
-        open(fd, off, len);
+    /**
+     * Initializes an ogg vorbis native file from a file descriptor [fd] of an already opened file.
+     * 
+     * @param fd an integer representing the file descriptor 
+     * @param offset an integer indicating the start of the buffer
+     * @param length an integer indicating the end of the buffer
+     * @throws IOException in cases of a failure to initialize the vorbis file 
+     */
+    public NativeVorbisFile(int fd, long offset, long length) throws IOException {
+        init(fd, offset, length);
     }
     
-    private native void open(int fd, long off, long len) throws IOException;
-    
+    /**
+     * Seeks to a playback time relative to the decompressed pcm (Pulse-code modulation) stream.
+     * 
+     * @param time the playback seek time
+     * @throws IOException if the seek is not successful
+     */
     public native void seekTime(double time) throws IOException;
     
-    public native int read(byte[] buf, int off, int len) throws IOException;
+    /**
+     * Reads the vorbis file into a primitive byte buffer [buf] with an [offset] indicating the read start and a [length] indicating the read end on the output buffer.
```

Comment:
Notice the documentation here is provisional and still subjected to modifications...